server/delay_server/src/irodsDelayServer.cpp

```diff
@@ -676,6 +677,51 @@ namespace
     } // make_delay_queue_query_processor
 } // anonymous namespace
 
+bool is_current_server_leader(void) {
+    std::string hostname;
+    char hn[HOST_NAME_MAX + 1]{};
+
+    if (gethostname(hn, sizeof(hn)) != 0) {
+        logger::delay_server::warn("is_current_server_leader: Failed to retrieve local server's hostname.");
+        return false;
+    }
+
+    if (const auto hn_resolved = resolve_hostname(hn, hostname_resolution_scheme::match_preferred); hn_resolved) {
+        hostname = *hn_resolved;
+    }
+    else {
+        hostname = std::string{hn};
+    }
+    auto conn = get_new_connection(std::nullopt);
+    std::string result;
+    {
+        GridConfigurationInput input{};
+        std::strcpy(input.name_space, "delay_server");
+        std::strcpy(input.option_name, "leader");
+
+        GridConfigurationOutput* output{};
+        irods::at_scope_exit free_output{[&output] { std::free(output); }};
+
+        if (const auto ec = rc_get_grid_configuration_value(static_cast<RcComm*>(conn), &input, &output); ec != 0) {
+            logger::delay_server::warn("is_current_server_leader: Failed to retriever leader config option.");
+            return false;
+        }
+
+        result = output->option_value;
+    }
+
+    std::string leader;
+    if (const auto hn_resolved = resolve_hostname(result, hostname_resolution_scheme::match_preferred); hn_resolved) {
```

Comment:
Should we check to see if `result` is empty?

Reply:
Before it is a check to see if `rc_get_grid_configuration_value` failed, and then after it is a check to see if `resolve_hostname` fails...I think that catches both cases where something could go awry; either `resolve_hostname` returns `std::nullopt` and it fails, or it was already broken before it...so I think it might be OK as is?

Reply:
Given the changes to this PR since this conversation happened, this can be marked as resolved.